### PR TITLE
AAP-7106 - Create document for P1 Repository Management features

### PR DESCRIPTION
**Pull request by dcdacosta**
Creating a new document for the Repository Management P1 features, including: 

- S1.1 [2.4 P1] Basic repository management for collections
- S3.1 [2.4 P1] Basic sync of single repository
- S4.1 [2.4 P1] RBAC on repositories
- S6.1 [2.4 P1] Ability to export/import collections

Preview link (VPN required): http://file.rdu.redhat.com/~ddacosta/AAP7106/master.html

### CODE REVIEW

**Review by newswangerd — APPROVED** [review]

**Review by hherbly — APPROVED** [review]
LGTM! I left one comment pointing out some missed capitalizations.

File: downstream/assemblies/hub/assembly-repo-management.adoc
[the remark this answers is not in the document]
Comment (jself-sudoku) [reply downstream/assemblies/hub/assembly-repo-management.adoc:1]: +1 @hherbly - agree.

File: downstream/assemblies/hub/assembly-repo-management.adoc [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+:context: repo-management
+
+[role="_abstract"]
+{HubNameStart} includes two types of repositories where you can publish content:
+
+Staging repositories:: any user with permission to upload for a namespace can publish content into these repositories. Content in these repositories are not available in the search page, but rather, are displayed on the approval dashboard for an administrator to verify.
+
+Custom repositories:: any user with write permissions on the repository can publish content to these repositories. These repositories are not displayed on the approval dashboard. They can show up on the search page depending on whether search is enabled by the owner of the repository. Custom repositories can be private repositories where only users with view permissions can see them, or public where all users can see them.
+
+Staging repositories are marked with the `pipeline=staging` label. By default {HubName} ships with one staging repository that is automatically used when a repository is not specified for uploading collections. However, users can create new staging repositories during xref:proc-create-repository[repository creation].
```
Comment (jself-sudoku) [inline downstream/assemblies/hub/assembly-repo-management.adoc:15]: Just a tiny nit: 
```suggestion
Staging repositories are marked with the `pipeline=staging` label. By default, {HubName} ships with one staging repository that is automatically used when a repository is not specified for uploading collections. However, users can create new staging repositories during xref:proc-create-repository[repository creation].
```

File: downstream/modules/automation-hub/con-approval-pipeline.adoc [hunk 2]
```diff
@@ -0,0 +1,20 @@
+// Module included in the following assemblies:
+// assembly-repo-management.adoc
+
+
+[id="con-approval-pipeline"]
+
+= Approval pipeline
+
+Automation Hub allows you to approve collections into any repository that is marked with the `pipeline=approved` label. By default, Hub ships with one repository for approved content, but you have the option to add more from the repository creation screen. Repositories marked with this label are not eligible for direct publishing and collections must come from one of the staging repositories.
```
Comment (jself-sudoku) [inline downstream/modules/automation-hub/con-approval-pipeline.adoc:9]: Missing attributes (one for the start of the sentence and one within a sentence; lower case mostly) . Do we have an attribute for just "Hub"? Might be good to add that to the list of attributes.

```suggestion
{HubNameStart} allows you to approve collections into any repository that is marked with the `pipeline=approved` label. By default, {HubName} ships with one repository for approved content, but you have the option to add more from the repository creation screen. Repositories marked with this label are not eligible for direct publishing and collections must come from one of the staging repositories.
```

File: downstream/modules/automation-hub/con-repo-rbac.adoc [hunk 3]
```diff
@@ -0,0 +1,9 @@
+// Module included in the following assemblies:
+// assembly-repo-management.adoc
+
+
+[id="con-repo-rbac"]
+
+= Role Based Access Conrol
```
Comment (jself-sudoku) [inline downstream/modules/automation-hub/con-repo-rbac.adoc:7]: Minor typo: "t" is missing.
```suggestion
= Role Based Access Control
```

File: downstream/modules/automation-hub/proc-provide-repository-access.adoc [hunk 4]
```diff
@@ -0,0 +1,20 @@
+// Module included in the following assemblies:
+// assembly-basic-repo-management.adoc
+
+[id="proc-provide-repository-access"]
+
+= Provide access to a repository
+
+By default, private repositories and the content are hidden from all users in the system. Public repositories can be viewed by all users, but can not be modified. Use this procedure to provide access to your custom repository.
```
Comment (jself-sudoku) [inline downstream/modules/automation-hub/proc-provide-repository-access.adoc:8]: Nit only:
```suggestion
By default, private repositories and the content are hidden from all users in the system. Public repositories can be viewed by all users, but cannot be modified. Use this procedure to provide access to your custom repository.
```

File: downstream/modules/automation-hub/proc-create-remote.adoc [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+====
++
+. Configure the credentials to the remote server by entering a *Token* or *Username* and *Password* required to access the external collection.
++
+[NOTE]
+====
+A Token can be generated by navigating to menu:Automation Hub[API token], clicking btn:[Load token] and copying the token that is loaded.
+====
++
+. To access collections from console.redhat.com, enter the *SSO URL* to sign in to the  identity provider (IdP).
```
Comment (jself-sudoku) [inline downstream/modules/automation-hub/proc-create-remote.adoc:29]: minor nit (extra space): 
```suggestion
. To access collections from console.redhat.com, enter the *SSO URL* to sign in to the identity provider (IdP).
```

File: downstream/modules/automation-hub/proc-create-remote.adoc [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+-----
++
+[NOTE]
+====
+All collection dependencies are automatically downloaded during the Sync process.
+====
++
+. To configure your remote further, use the options available under *Advanced configuration*:
+.. If there is a corporate proxy in place for your organization, enter a *Proxy URL*, *Proxy Username* and *Proxy Password*.
+.. Enable or disable transport layer security using *TLS validation* checkbox.
```
Comment (jself-sudoku) [inline downstream/modules/automation-hub/proc-create-remote.adoc:46]:
```suggestion
.. Enable or disable transport layer security using the *TLS validation* checkbox.
```

File: downstream/assemblies/hub/assembly-repo-sync.adoc [hunk 7]
```diff
@@ -0,0 +1,14 @@
+ifdef::context[:parent-context: {context}]
+
+[id="repository-sync"]
+= Repository sync
+
+:context: repository-sync
+
+[role="_abstract"]
+You can distribute relevant collection content to your users by synchronizing repositories from one automation hub to another. You should synchronize your custom repository with the remote to ensure you have the latest collection updates.
```
Comment (jself-sudoku) [inline downstream/assemblies/hub/assembly-repo-sync.adoc:9]:
```suggestion
You can distribute relevant collection content to your users by synchronizing repositories from one {HubName} to another. You should synchronize your custom repository with the remote to ensure you have the latest collection updates.
```

File: downstream/modules/automation-hub/proc-basic-repo-sync.adoc [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+// assembly-repo-sync.adoc
+
+[id="proc-basic-repo-sync"]
+
+= Basic repository sync
+
+Use this procedure to sync a repository from one automation hub instance to another.
+
+.Procedure
+. Log in to Red Hat Ansible Automation platform.
```
Comment (jself-sudoku) [inline downstream/modules/automation-hub/proc-basic-repo-sync.adoc:11]:
```suggestion
. Log in to {PlatformName}.
```

**Comment by jself-sudoku** [comment]
Other than a few missed attributes, this LGTM @dcdacosta. Cohesive and easy for customers to understand.

File: downstream/assemblies/hub/assembly-repo-management.adoc
[the remark this answers is not in the document]
Comment (dcdacosta) [reply downstream/assemblies/hub/assembly-repo-management.adoc:1]: Fixed.

File: downstream/assemblies/hub/assembly-repo-management.adoc (hunk 1 above)
Comment (dcdacosta) [reply downstream/assemblies/hub/assembly-repo-management.adoc:15]: Comma added.

File: downstream/assemblies/hub/assembly-repo-sync.adoc (hunk 7 above)
Comment (dcdacosta) [reply downstream/assemblies/hub/assembly-repo-sync.adoc:9]: Changed to attribute.

File: downstream/modules/automation-hub/con-approval-pipeline.adoc (hunk 2 above)
Comment (dcdacosta) [reply downstream/modules/automation-hub/con-approval-pipeline.adoc:9]: Changed to attributes.

File: downstream/modules/automation-hub/con-repo-rbac.adoc (hunk 3 above)
Comment (dcdacosta) [reply downstream/modules/automation-hub/con-repo-rbac.adoc:7]: Fixed.

File: downstream/modules/automation-hub/con-repo-rbac.adoc [hunk 9]
```diff
@@ -0,0 +1,9 @@
...
+
+Role Based Access Control (RBAC) restricts user access to custom repositories based on their defined role. By default, users can view all public repositories in their {HubName}, but they can not modify them unless they have explicit permission to do so. This is the same for other operations on the repository. For example, removing a user's rights revokes their ability to download content from a custom repository. See link:https://access.redhat.com/documentation/en-us/red_hat_ansible_automation_platform/2.3/html/managing_user_access_in_private_automation_hub/assembly-user-access[Configuring user access for your local Automation Hub] for information about managing user access in automation hub.
```
[the remark this answers is not in the document]
Comment (dcdacosta) [reply downstream/modules/automation-hub/con-repo-rbac.adoc:9]: Fixed.

File: downstream/modules/automation-hub/proc-basic-repo-sync.adoc (hunk 8 above)
Comment (dcdacosta) [reply downstream/modules/automation-hub/proc-basic-repo-sync.adoc:11]: Fixed.

File: downstream/modules/automation-hub/proc-create-remote.adoc (hunk 5 above)
Comment (dcdacosta) [reply downstream/modules/automation-hub/proc-create-remote.adoc:29]: Fixed.

File: downstream/modules/automation-hub/proc-create-remote.adoc (hunk 6 above)
Comment (dcdacosta) [reply downstream/modules/automation-hub/proc-create-remote.adoc:46]: Fixed.

File: downstream/modules/automation-hub/proc-provide-repository-access.adoc (hunk 4 above)
Comment (dcdacosta) [reply downstream/modules/automation-hub/proc-provide-repository-access.adoc:8]: Fixed.

File: downstream/modules/automation-hub/proc-provide-repository-access.adoc [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+
+.Procedure
+. Log in to {PlatformName}.
+. Navigate to menu:Automation Hub[Repositories].
+. Locate your repository in the list and click image:more_actions.png[more actions], then select *Edit*.
+. Select the *Access* tab.
+. Select a group for *Repository owners*.
++
+See link:https://access.redhat.com/documentation/en-us/red_hat_ansible_automation_platform/2.3/html/managing_user_access_in_private_automation_hub/assembly-user-access[Configuring user access for your local Automation Hub] for information about implementing user access.
```
[the remark this answers is not in the document]
Comment (dcdacosta) [reply downstream/modules/automation-hub/proc-provide-repository-access.adoc:17]: Good catch! Will fix once 2.4 is published as I'm not confident that the only change to the link is the version number.

**Review by hherbly — APPROVED** [review]

**Comment by hherbly** [comment]
LGTM. I can squash and merge once it's out of draft form. :)

**Pull request merged**